File: go/solutions/0171-excel-sheet-column-number.go
package solutions

import "math"

func TitleToNumber(columnTitle string) int {
	var num int
	j := len(columnTitle) - 1
	for i := 0; i < len(columnTitle); i++ {
		num += (int(columnTitle[i]-'A') + 1) * (int(math.Pow(26, float64(j))))
		j--
	}

	return num
}

File: js/0171-excel-sheet-column-number.js
/**
 * @param {string} columnTitle
 * @return {number}
 */
var titleToNumber = function (columnTitle) {
  var num = 0,
    j = columnTitle.length - 1;
  for (var i = 0; i < columnTitle.length; i++) {
    num +=
      (columnTitle.charCodeAt(i) - "A".charCodeAt(0) + 1) * Math.pow(26, j);
    j--;
  }

  return num;
};

module.exports = { titleToNumber };
